addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxCubeBridgeDiscovery.java

```diff
@@ -85,6 +77,148 @@ public boolean isBackgroundDiscoveryEnabled() {
 		return true;
 	}
 
+	private synchronized void discoverCube() {
+		discoveryRunning = true;
+		Thread thread = new Thread("Sendbroadcast"){
+			public void run(){
+				sendDiscoveryMessage (MAXCUBE_DISCOVER_STRING);
+				try {
+					sleep(5000);
+				} catch (Exception e) {
+				}
+				discoveryRunning = false;
+				logger.trace( "Done sending broadcast discovery messages.");
+			}
+		};
+		thread.start();
+		receiveDiscoveryMessage();
```

Comment:
Hi @kaikreuzer 
Yes, can fix that. I can't just move it in the thread above as that thread is making sure that the receive process (receiveDiscoveryMessage()) is stopped after a while (with the discoveryRunning = false;). Maybe I just start it in a separate thread. Anyway, I'll find something and submit a PR